blazor/datagrid/filtering.md

```diff
@@ -179,7 +179,11 @@ In the following sample, the `SfDropDownList` component is rendered in the [Filt
         <GridColumn Field=@nameof(Order.Freight) HeaderText="Freight" Format="C2" TextAlign="TextAlign.Right" Width="120"></GridColumn>
         <GridColumn Field=@nameof(Order.Type) HeaderText="Type" Type="ColumnType.String" Width="130">
             <FilterTemplate>
-                <SfDropDownList Placeholder="Type" ID="Type" Value="@((string)(context as PredicateModel).Value)" DataSource="@FilterDropData" TValue="string" TItem="Data">
+                 @{
+                    var enumValue = (context as PredicateModel).Value;
+                    var filterValue = enumValue == null ? string.Empty : enumValue.ToString();
```

Comment:
explain the reason why was the variable used. any exception occur in previous way?

Reply:
Exception thrown. When we try to reload or refresh the page while enabling persistence, the current filtered value is got as a string. So unable to cast an object type to string.

 Hence, we need to store the predicate model in a variable, and filtervalue is converted to a string in the sample level using the ToString() method.

Reply:
we have not enabled the persistence in that sample right. then what is the purpose?